pkg/icons/icons.go

```diff
@@ -0,0 +1,80 @@
+package icons
+
+import (
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/rancher/charts-build-scripts/pkg/filesystem"
+	"github.com/rancher/charts-build-scripts/pkg/path"
+	"github.com/sirupsen/logrus"
+	helmRepo "helm.sh/helm/v3/pkg/repo"
+)
+
+// DownloadIcons download icons to use locally
+func DownloadIcons(rootFs billy.Filesystem) error {
+	absRepositoryHelmIndexFile := filesystem.GetAbsPath(rootFs, path.RepositoryHelmIndexFile)
+	helmIndexFile, err := helmRepo.LoadIndexFile(absRepositoryHelmIndexFile)
+	if err != nil {
+		return fmt.Errorf("encountered error while trying to load existing index file: %s", err)
+	}
+	helmIndexFile.SortEntries()
+	for name := range helmIndexFile.Entries {
+		latestEntryVersion := helmIndexFile.Entries[name][0]
+		latestIconName := name
+		if strings.Contains(name, "-crd") {
+			continue
+		}
+		latestIcon := latestEntryVersion.Icon
+		if latestIcon == "" || strings.HasPrefix(latestIcon, "file://") {
+			logrus.Infof("Skipping icon download for %s", name)
+			continue
+		}
+		p := fmt.Sprintf("assets/logos/%s", latestIconName)
+		logrus.Infof("Downloading icon download for %s", name)
+		err := download(rootFs, latestIcon, p)
+		if err != nil {
+			logrus.Infof("failed to download icon for %s", name)
+			continue
```

Comment:
@rohitsakala 
We don't want to return an error because sometimes one Icon URL might be broken.
Then we should keep the icon field with the latest successful icon file.

Of course, we will need to do a one-time effort to guarantee we have a single available icon file for all charts. 

From this point on, if the URL is broken we don't need to halt the flow because we still have one icon working.

Reply:
Okay, gotcha. So we make sure all icons are present first.